### ISSUE TITLE
Known problems with Docker 1.7 client

### ISSUE THREAD

**Issue by hjacobs** [zalando-stups/pierone#24]
Docker 1.7 gets EOF while trying to start containers from Pier One, exact reason is unknown.

Workaround: Downgrade to Docker 1.6.2, on Ubuntu:

```
sudo apt-get install lxc-docker-1.6.2
```

**Comment by elgalu** [comment]
Same pain here. I wasn't able to rollback to 1.6.2 successfully however below steps worked for me to get 1.7.0 working on Ubuntu 14.04

``` sh
# 1. completely uninstall docker
sudo service docker stop && sudo apt-get purge lxc-docker lxc-docker-1.7.0 lxc
sudo ifconfig docker0 down && sudo brctl delbr docker0 && sudo groupdel docker
# DANGER: below code will remove all images and containers
sudo rm -rf /var/lib/docker /dev/mapper/docker* /usr/bin/docker /run/docker /etc/docker /etc/default/docker /etc/init.d/docker /etc/apparmor.d/cache/docker /etc/apparmor.d/docker /etc/bash_completion.d/docker /var/log/upstart/docker* /var/lib/apt/lists/*docker* /var/lib/dpkg/info/*docker* /dev/disk/by-id/*docker*
# 2. remove kernel 3.18.16 and go back to 3.13.0-55
sudo apt-get purge 'linux-headers*' 'linux-image*'
sudo apt-get install linux-headers-3.13.0-55 linux-headers-3.13.0-55-generic linux-image-extra-3.13.0-55-generic linux-image-3.13.0-55-generic
sudo update-grub && sudo reboot
# 3. reinstall latest docker 1.7.0
sudo apt-get install aufs-tools
curl -s https://get.docker.io/ubuntu/ | sudo sh
sudo gpasswd -a $USER docker && sudo reboot
```

Final result:

``` sh
docker --version
#=> Docker version 1.7.0, build 0baf609

docker info
#=> Containers: 0 Images: 94
#=> Storage Driver: aufs  Root Dir: /var/lib/docker/aufs
#=>  Backing Filesystem: extfs Execution Driver: native-0.2
#=> Logging Driver: json-file  Kernel Version: 3.13.0-55-generic
#=> Operating System: Ubuntu 14.04.2 LTS
```

**Comment by hjacobs** [comment]
container config JSON diffs (notworking.json was created by downloaded from Pier One, working.json was created locally only):

```
$ diff notworking.json working.json 
29c29
<   "Image": "e837d5a2bbf7f31f5e5ee21ff946618cf139d69c2a415001a431e8f703c5b8bb",
---
>   "Image": "817517b30c1fbf3a28f45c582df7080d1b423934941d458be6e7bed52baa087d",
38c38
<       "tcp": {}
---
>       "8080/tcp": {}
46c46
<     "Hostname": "b36fa6b38fab",
---
>     "Hostname": "2aa74daf401a",
58c58
<     "Image": "e837d5a2bbf7",
---
>     "Image": "817517b30c1f",
67,68c67,68
<   "Created": "2015-06-24T09:34:18.474255626Z",
<   "ID": "b36fa6b38fabadec8568a08eaca3203ac3a875dfe2d7ec478acffb29682141d1",
---
>   "Created": "2015-06-24T09:36:02.962665473Z",
>   "ID": "2aa74daf401acec409618f377c850433db67f808e5b91309b4492cf5db6be063",
85c85
<   "Name": "/pensive_payne6",
---
>   "Name": "/gloomy_heisenberg",
```

Note the difference in "ExposedPorts".

**Comment by hjacobs** [comment]
fixed in release 0.6.0 https://github.com/zalando-stups/pierone/releases/tag/0.6.0
